helix-term/src/commands.rs

```diff
@@ -638,11 +640,12 @@ fn goto_window(cx: &mut Context, align: Align) {
     let last_line = view.last_line(doc);
 
     let line = match align {
-        Align::Top => (view.offset.row + scrolloff),
-        Align::Center => (view.offset.row + (height / 2)),
-        Align::Bottom => last_line.saturating_sub(scrolloff),
+        Align::Top => (view.offset.row + scrolloff + count),
+        Align::Center => (view.offset.row + ((last_line - view.offset.row) / 2)),
+        Align::Bottom => last_line.saturating_sub((scrolloff + count).min(last_line)),
```

Comment:
```suggestion
        Align::Bottom => last_line.saturating_sub((scrolloff + count)),
```

`saturating_sub` should already protect against overflow here.